biahub/cli/slurm.py

```diff
@@ -22,3 +30,136 @@ def wait_for_jobs_to_finish(jobs: list[submitit.Job]) -> None:
             pass  # as_completed polls every 10 seconds by default, so we don't need to do anything here
         except Exception as e:
             print(f"Job {job.job_id} failed with exception: {e}")
+
+
+def check_job_logs(
+    log_dir: Union[str, Path],
+    job_ids: Optional[list[Union[str, int]]] = None,
+    output_dir: Union[str, Path] = None,
+) -> pd.DataFrame:
+    """
+    Check SLURM logs for specific job IDs or all jobs in a folder,
+    reporting statuses and saving a filtered CSV with non-successful jobs.
+
+    Parameters
+    ----------
+    log_dir : str or Path
+        Path to the SLURM log directory.
+    job_ids : list[str or int], optional
+        Specific job IDs to check. If None, will scan all *_log.out files in the directory.
+
+    Returns
+    -------
+    df_filtered : pd.DataFrame
+        DataFrame with only non-success job statuses.
+    """
+    log_dir = Path(log_dir)
+    records = []
+
+    if job_ids is None:
+        job_ids = sorted(
+            [
+                fname.name.replace("_log.out", "").replace("_log.err", "")
+                for fname in log_dir.glob("*_log.out")
+            ]
+        )
+
+    for job_id in tqdm(job_ids, desc="Analyzing SLURM logs"):
+        job_id = str(job_id)
+
+        all_files = os.listdir(log_dir)
+        matching_files = [f for f in all_files if f"{job_id}" in Path(f).stem]
+
+        files_with_out = [f for f in matching_files if f.endswith("_log.out")]
+
+        for file_out in files_with_out:
+            log_out = log_dir / file_out
+            log_err = log_dir / file_out.replace("_log.out", "_log.err")
+            id = Path(log_err).stem.replace("_0_log", "")  # safer
+
+            if not log_out.exists() and not log_err.exists():
+                status = "LOG NOT FOUND"
+            else:
+                try:
+                    logs_combined = ""
+                    if log_out.exists():
+                        logs_combined += log_out.read_text().lower()
+                    if log_err.exists():
+                        logs_combined += log_err.read_text().lower()
+
+                    if (
+                        "job completed successfully" in logs_combined
+                        or "exiting after successful completion" in logs_combined
+                    ):
+                        status = "SUCCESS"
+                    else:
+                        status = "FAILED"
+                except Exception as e:
```

Comment:
Bare exceptions can be uninformative. Do we know what kinds of errors would show up here? Would a `FileNotFound` make sense, or maybe an `OSError`?

Here's a list of default [Python exceptions][py-errors].

[py-errors]: https://docs.python.org/3/library/exceptions.html